Source/DafnyDriver/SynchronousCliCompilation.cs

```diff
@@ -306,7 +308,15 @@ public class SynchronousCliCompilation : IDisposable {
           compiled = false;
         }
 
-        exitValue = verified && compiled ? ExitValue.SUCCESS : !verified ? ExitValue.VERIFICATION_ERROR : ExitValue.COMPILE_ERROR;
+        var postCompilationErrorCount = dafnyProgram.Reporter.ErrorCount;
```

Comment:
These changes are similar to those in this PR: https://github.com/dafny-lang/dafny/pull/4971/files#diff-45175daf8157d0a1d1f2f7ed84bff061902f63276e5066c6d191c30af5004157R309

Would you mind if we merge that one first? I believe it achieves the same as these changes, but also does more when there are warnings.

Reply:
Oh, I hadn't seen that PR! It definitely seems better.